packages/blade/src/components/Input/BaseInput/__tests__/BaseInput.web.test.tsx

```diff
@@ -0,0 +1,177 @@
+import userEvent from '@testing-library/user-event';
+
+import { BaseInput } from '..';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.web';
+import ThemeWrapper from '~src/_helpers/testing/themeWrapper';
+import assertAccessible from '~src/_helpers/testing/assertAccessible.web';
+import { CloseIcon, EyeIcon } from '~components/Icons';
+
+describe('<BaseInput />', () => {
+  it('should render', () => {
+    const { container } = renderWithTheme(<BaseInput label="Enter name" id="name" />);
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should display success validation state', () => {
+    const label = 'Enter name';
+    const { getByText, getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" validationState="success" successText="Success" />,
+    );
+
+    const input = getByLabelText(label);
+    const successText = getByText('Success');
+
+    expect(successText).toBeTruthy();
+    expect(input).toHaveAccessibleDescription('Success');
+    expect(input).toBeValid();
+  });
+
+  it('should display error validation state', () => {
+    const label = 'Enter name';
+    const { getByText, getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" validationState="error" errorText="Error" />,
```

Comment:
In all 3 tests testing validationState let's add all the props. helpText/successText/errorText, this will ensure that our component isn't picking some other text when validation state is set.

Reply:
Added in https://github.com/razorpay/blade/pull/695/commits/8be6b8367acaeaf5f157a5ba38eb97ff0c75f785